### PR TITLE
[CoSim] cpp ping pong

### PR DESCRIPTION
**Pull request by adityaghantasala**
This Adds the Ping-Pong as a test case to the co-sim app.

Ping and Pong are made both cpp solvers independent of the Kratos framework. 
The corresponding ping_pong_wrapper.py and ping_pong_io.py are also added. 
The solvers Ping and Pong exchange a double between them via CoSimApp ! 
I will add an iterative exchange in the next PR. 

This demonstrates the usage of the co simulation app to couple two solver which are interdependent of the kratos framework .

### CODE REVIEW

**Comment by loumalouomega** [comment]
Any video?, I would like to see ping pong stuff

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/cpp_ping_pong/ping_pong_io.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    def ExportCouplingInterface(self, interface_config):
+        pass
+
+    def ImportData(self, data_config):
+        data_type = data_config["type"]
+        if data_type == "coupling_interface_data":
+            interface_data = data_config["interface_data"]
+            data_name = self.solver_name+"_"+interface_data.name
+            data_file_name = "EMPIRE_datafield_" + data_name + ".dat"
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/cpp_ping_pong/ping_pong_io.py:25]: why don't you use the CoSim- EMPIRE_API directly?

**Comment by philbucher** [comment]
> Any video?, I would like to see ping pong stuff

It unfortunately is not what the name implies :sweat_smile: 
PingPong in the cosim sense typically means the exchange of a scalar between two solvers to show that the coupling works :)

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/cpp_ping_pong/ping_pong_io.py (hunk 1 above)
Comment (adityaghantasala) [reply applications/CoSimulationApplication/python_scripts/solver_wrappers/cpp_ping_pong/ping_pong_io.py:25]: as I said .. I want to show something else can also be done and it need not use what is existing ! 

But ya EMPIRE_API also will do the job !

**Comment by loumalouomega** [comment]
> > Any video?, I would like to see ping pong stuff
> 
> It unfortunately is not what the name implies sweat_smile
> PingPong in the cosim sense typically means the exchange of a scalar between two solvers to show that the coupling works :)

imagen

**Review by philbucher — DISMISSED** [review]

**Comment by adityaghantasala** [comment]
@philbucher  can you approve this again .. I made the changes we discussed !

**Review by philbucher — DISMISSED** [review]
this will probably require some minor updates after #5543 but we can do it later

**Comment by philbucher** [comment]
codacy has some complaints, could you fix those before merging?

**Comment by adityaghantasala** [comment]
Done !

**Review by philbucher — APPROVED** [review]

**Pull request merged**